services/cd-service/pkg/repository/transformer.go

```diff
@@ -121,11 +122,37 @@ func UpdateDatadogMetrics(state *State) error {
 		return err
 	}
 	for env := range configs {
-		GaugeEnvLockMetric(fs, env)
-		appsDir := fs.Join(environmentDirectory(fs, env), "applications")
-		if entries, _ := fs.ReadDir(appsDir); entries != nil {
+		GaugeEnvLockMetric(filesystem, env)
+		appsDir := filesystem.Join(environmentDirectory(filesystem, env), "applications")
+		if entries, _ := filesystem.ReadDir(appsDir); entries != nil {
 			for _, app := range entries {
-				GaugeEnvAppLockMetric(fs, env, app.Name())
+				GaugeEnvAppLockMetric(filesystem, env, app.Name())
+			}
+		}
+	}
+	now := time.Now() // ensure all events have the same timestamp
+	if changes != nil && ddMetrics != nil {
+		for i := range changes.ChangedApps {
+			oneChange := changes.ChangedApps[i]
+			teamMessage := func() string {
+				if oneChange.Team != "" {
+					return fmt.Sprintf(" for team %s", oneChange.Team)
+				}
+				return ""
+			}()
+			event := statsd.Event{
+				Title:     "Kuberpult app deployed",
+				Text:      fmt.Sprintf("Kuberpult has deployed %s to %s%s", oneChange.App, oneChange.Env, teamMessage),
+				Timestamp: now,
+				Tags: []string{
+					"app:" + oneChange.App,
```

Comment:
Maybe rather `kuberpult.application`, `kuberpult.environment` and `kuberpult.team`? The tags are very generic at the moment and datadog is weird when the tags collide.

Reply:
probably better, yes.

Reply:
done